feature_engine/base_transformers.py

```diff
@@ -144,3 +144,26 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
     # for the check_estimator tests
     def _more_tags(self):
         return _return_tags()
+
+    def get_feature_names(
+        self,
+        input_features: List[str] = None,
```

Comment:
currently, feature-engine also supports integers as column names, see for example [this](https://github.com/solegalli/feature_engine/blob/master/feature_engine/imputation/mean_median.py#L64).

Reply:
Thanks for the reference, I've fixed it here: c784f6d311a4e5dafd5d18ea5a40ceefa2702e2c.

I'm following the implementation in scikit's OneHotEncoder as an example https://github.com/scikit-learn/scikit-learn/blob/efede4b1864492c18238f60ffc39e9729672382c/sklearn/preprocessing/_encoders.py#L658. 

From what I understand, the `input_features` argument servers to limit the desired output features by the ones that directly map from features included. For example, `feature_engine.OneHotEncoder` can be fitted to encode multiple columns - calling `get_feature_names()` on the fitted encoder would return all transformed columns (`column_1 * cats_in_column_1 + column_2 * cats_in_column_2 ...`); however calling `get_feature_names(input_features=['column_1'])` would just return the features derived from that column: `column_1 * cats_in_column_1 `.

This is not essential for compatibility with eli5, I don't think. My intention was to follow a reference implementation of `get_feature_names`

Reply:
I see. My understanding is that sklearn has this functionality, because as it returns numpy arrays, you more or less have to guess which feature comes from which original variable. There is similar behaviour with the PolynomialFeatures class. So to help the user they created this method.

Feature-engine does not have that problem. If new variables are added, their name starts with the original variable name. Or alternatively, the user provides the names. So you always know what you get.

So thinking it further, I am not even sure we need a method here. If we were to store the name and order of the original variables, as they appeared in the training set, we would just need to create a new attribute, `feature_names_` in all transformers, and add it to the doscstrings.

But I need more convincing that this is useful / urgent :p

Reply:
I don't think pre-calculating the feature names is a problem, or even storing it as a new attribute.  
I would only argue that `get_feature_names` is currently the expected API call for this - even if it is looking like it is going to change in the future. 

My use-case - eli5 - for example,  expects this as the way to retrieve the names of the features. But also pipelines are composable, so if one uses feature_engine encoders alongside ColumnTransformers - it is expected that features will be able to be retrieved in the same form: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/compose/_column_transformer.py#L372

I'm not sure I can make an honest argument on urgency, based on my own usecase :)

I can point to about [100k code results](https://github.com/search?q=get_feature_names&type=code) for the function name, [~250 issues](https://github.com/search?q=get_feature_names&type=issues) across multiple projects, and a [5-year-old](https://github.com/scikit-learn/scikit-learn/issues/6425) effort to unify the behaviour in scikit-learn

Reply:
OK then, I won't oppose any further :)

Thanks a lot for the links and the initiative.

Would it be ok to make the method so that it takes no argument?

I think for most transformers list(X.columns) should do. 

Special treatment would be required for the transformers in the modules creation, selection and wrappers and for the one hot encoder, of the top of my head.

Shall I leave this with you?

Reply:
681bc8a

Some tests are failing on the one hot encoder. I think it's because there's an inconsistency on columns that should get dropped